internal/pipeline/executor.go

```diff
@@ -308,8 +308,20 @@ func hydrate(p *pkg, hctx *hydrationContext) (resources []*yaml.RNode, err error
 func filterMetaData(resources []*yaml.RNode) []*yaml.RNode {
 	var filtered []*yaml.RNode
 	for _, r := range resources {
-		meta, _ := r.GetMeta()
-		if !strings.Contains(meta.APIVersion, "kpt.dev") {
+		m, _ := r.GetMeta()
+		if !strings.Contains(m.APIVersion, "kpt.dev") {
+			filtered = append(filtered, r)
+		}
+	}
+	return filtered
+}
+
+// filterFnConfig filters out function configs
+func filterFnConfig(resources []*yaml.RNode) []*yaml.RNode {
```

Comment:
This is definitely needed. pl. see my comment above about additional filtering. Feel free to do that in separate PR if that helps.